akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -464,8 +518,10 @@ class ClusterSharding(system: ExtendedActorSystem) extends Extension {
    */
   def shardRegionProxy(typeName: String, dataCenter: DataCenter): ActorRef = {
     regions.get(proxyName(typeName, Some(dataCenter))) match {
-      case null ⇒ throw new IllegalArgumentException(s"Shard type [$typeName] must be started first")
-      case ref  ⇒ ref
+      case null ⇒
```

Comment:
Shouldn't that be `regions.get(proxyName(s"${typeName}Proxy", Some(dataCenter)))` here as well?

Reply:
Would it be convenient to split `regions` into `regions` and `proxies`?

Reply:
I agree with @raboof. There is still something that is not quite right here. I'll add some more comments.

Reply:
I think it's a good suggestion to add another `proxies: ConcurrentHashMap` and then just use the `proxyName(typeName, Some(dataCenter))` everywhere when accessing that map.